### PR TITLE
refactor: Ditch the `Mock` folder

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## What is the current behavior?

Some of the files have their `.xyz.cs` portions inside the Mock folder, which makes it confusing when looking for them.

## What is the new behavior?

Remove mock folder, delete unused files and move relevant ones to appropriate locations

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Media/ImageSource.wasm.cs [hunk 1]
```diff
@@ -19,6 +19,11 @@ namespace Microsoft.UI.Xaml.Media
 {
 	partial class ImageSource
 	{
+		public ImageSource()
+		{
+			InitializeBinder();
+		}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Media/ImageSource.wasm.cs:25]: I see it's pre-existing, but I think the `InitializeBinder` is not needed at all.

**Comment by jeromelaban** [comment]
@MartinZikmund looks like there's a breaking change to ignore for this one.

**Comment by Youssef1313** [comment]
The breaking changes here don't feel safe. Adding them to PackageDiffIgnore will still likely fail `ReferenceImplComparer`.

**Comment by MartinZikmund** [comment]
> The breaking changes here don't feel safe. Adding them to PackageDiffIgnore will still likely fail `ReferenceImplComparer`.

yes, I will re-add the missing things to avoid breaking changes

**Review by Youssef1313 — APPROVED** [review]

**Pull request merged**